packages/tsup-configs/src/dev.ts

```diff
@@ -0,0 +1,18 @@
+import { type DefineConfigOptions, defineConfigBase } from "./defineConfigBase.ts";
+
+
+export function defineDevConfig(config?: DefineConfigOptions) {
+    return defineConfigBase(config, {
```

Comment:
Instead of using `defineConfigBase`, I would use https://www.npmjs.com/package/deepmerge. This is the package used by `@squide` and the package that I believe should be used by all our libraries needing to merge 2 object literals.

Reply:
Got it, then I woud rename `defineConfigBase` to something like `mergeConfigs`